tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php

```diff
@@ -165,7 +165,7 @@ public function testDontFinishNoCache() : void
 
         $this->hydrateStmt($stmt, FetchMode::NUMERIC);
 
-        self::assertCount(2, $this->sqlLogger->queries);
+        self::assertCount(1, $this->sqlLogger->queries);
```

Comment:
Wait, isn't this test supposed to have 2 queries executed?

Reply:
Not really, a second query he took from the cache

Reply:
It previously expected 2 queries in the test and I want to know what's the reason for this test. I'll need help from @beberlei, who implemented this test in #69 a long time ago.

Reply:
The test was incorrect because, just will save the query in $this->sqlLogger->queries when the query are execute in the database.

https://github.com/doctrine/dbal/blob/4c03ed81471c62f178581eb325339a3c34f3b71a/lib/Doctrine/DBAL/Connection.php#L894-L897

When the query are in the cache, they return before

https://github.com/doctrine/dbal/blob/4c03ed81471c62f178581eb325339a3c34f3b71a/lib/Doctrine/DBAL/Connection.php#L886-L897

The reason are the cache doesn't work before, when we use fetch

Reply:
I agree, which is why I approved if at first, but if there's another reason why @beberlei expected 2 in the past, I want to know and would like to be on the save side.

Reply:
This was recently changed in https://github.com/doctrine/dbal/pull/4048. It's an improvement, not a bugfix, so this change shouldn't be made in `2.10.x.`.